src/main/java/bowling/domain/frame/Frame.java

```diff
@@ -0,0 +1,69 @@
+package bowling.domain.frame;
+
+import bowling.domain.Pins;
+import bowling.domain.PitchingResults;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static bowling.domain.frame.NormalFrame.NORMAL_FRAME_MAXIMUM_INDEX;
+
+public abstract class Frame {
+    private final int index;
+    protected final PitchingResults pitchingResults;
+    private Frame nextFrame;
+
+    protected Frame(final int index) {
+        this.index = index;
+        this.pitchingResults = PitchingResults.of();
+    }
+
+    public static Frame createFirst() {
+        return NormalFrame.of(0);
+    }
+
+    public Frame createNext() {
+        if (index == NORMAL_FRAME_MAXIMUM_INDEX) {
+            nextFrame = FinalFrame.of(index + 1);
+            return nextFrame;
+        }
+
+        nextFrame = NormalFrame.of(index + 1);
+        return nextFrame;
+    }
+
+    public List<Frame> toFrames() {
+        if (this.index != 0) {
+            throw new IllegalStateException();
+        }
+        final List<Frame> frames = new ArrayList<>(10);
+        Frame frame = this;
+        frames.add(frame);
+        while (frame.next() != null) {
```

Comment:
```suggestion
        while (frame.nonLast()) {
```
다음 프레임이 null 임을 체크하는 것 보다는 메서드를 통해 의미를 부여해보는 건 어떨까요?
가독성이 좀 더 좋아질 수 있을것 같아요! 😄